File: nest-contract-test-example/src/customer/customer.controller.ts
import { Body, Controller, Get, Post } from '@nestjs/common';
import { CustomerService } from './customer.service';
import { CustomerRequest } from './customer.dto';

@Controller('customer')
export class CustomerController {
  constructor(private readonly customerService: CustomerService) {}

  @Get()
  findAll() {
    return this.customerService.findAll();
  }

  @Post()
  create(@Body() body: CustomerRequest) {
    return this.customerService.create(body);
  }
}

File: kotlin-contract-test-example/src/main/kotlin/com/example/kotlincontracttestexample/web/CustomerController.kt
package com.example.kotlincontracttestexample.web

import com.example.kotlincontracttestexample.service.example.CustomerRequest
import com.example.kotlincontracttestexample.service.example.CustomerResponse
import com.example.kotlincontracttestexample.service.example.CustomerService
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/customer")
class CustomerController(private val customerService: CustomerService) {
    @GetMapping
    fun findAll(): List<CustomerResponse> {
        return customerService.findAll()
    }

    @PostMapping
    fun create(@Validated @RequestBody body: CustomerRequest): CustomerResponse {
        return customerService.create(body)
    }
}